autoscheduler/frontend/src/components/SchedulingPage/CourseSelectColumn/CourseSelectCard/ExpandedCourseCard/ExpandedCourseCard.tsx

```diff
@@ -45,35 +44,32 @@ const ExpandedCourseCard: React.FC<ExpandedCourseCardProps> = ({
     });
   }
 
-  // determine customization content based on customization level
-  let customizationContent: JSX.Element = null;
-  switch (customizationLevel) {
-    case CustomizationLevel.BASIC:
-      customizationContent = <BasicSelect id={id} />;
-      break;
-    case CustomizationLevel.SECTION:
-      customizationContent = course
-        ? <SectionSelect id={id} />
-        : (
-          <Typography className={styles.grayText}>
-            Select a course to show available sections
-          </Typography>
-        );
-      break;
-    default:
-      customizationContent = null;
-  }
-
-  // show loading if we're not sure what sections are available
-  if (loading) {
-    customizationContent = (
-      <div id={styles.centerProgress}>
-        <SmallFastProgress />
-        Loading sections...
-      </div>
-    );
-  }
-  React.useEffect(() => setLoading(false), [sections]);
+  // determine customization content based on whether the card is loading and customization level
+  const customizationContent = React.useMemo(() => {
```

Comment:
I definitely noticed that it's faster now, but honestly I think that's mostly due to me checking if the card is loading, rather than from the `useMemo`, since now that I look at this again it will have to recalculate `customizationContent` most of the time when a variable changes. I'll try to think of a more scientific method of whether the memoization is helping and post my results here when I do.

Reply:
So I've done some testing for this now, as a "benchmark" I looked at the react profiler with/without the `useMemo` and did 2 things:
1. Made a bunch of cards with good ol' MATH 151, reloaded, and looked at render times whenever each card loaded. With the `useMemo` they averaged around 200ms, and without it was around 400ms.
2. Typed in MATH 151 and clicked on the result after it loaded. With `useMemo` it took around 400ms to render, without it divided the render into 3 events that were around 600, 1200, and 600 ms
So this is definitely a good use case for `useMemo`

Reply:
美しい